inject_test.go

```diff
@@ -114,101 +123,182 @@ func TestInjectClientsPanics(t *testing.T) {
 		},
 		{
 			name: "unknown service non-nil",
-			target: &struct {
-				Client json.Client `service:"foo"`
-			}{
-				Client: json.New(clientconfig.MultiOutbound(
-					"foo",
-					"bar",
-					transport.Outbounds{
-						Unary: transporttest.NewMockUnaryOutbound(mockCtrl),
-					})),
+			build: func(t *testing.T, mockCtrl *gomock.Controller) (tt testCase) {
+				tt.target = &struct {
+					Client json.Client `service:"foo"`
+				}{
+					Client: json.New(clientconfig.MultiOutbound(
+						"foo",
+						"bar",
+						transport.Outbounds{
+							Unary: transporttest.NewMockUnaryOutbound(mockCtrl),
+						})),
+				}
+				tt.wantNonNil = []string{"Client"}
+				return
 			},
-			wantNonNil: []string{"Client"},
 		},
 		{
 			name: "unknown type untagged",
-			target: &struct {
-				Client unknownClient `notservice:"foo"`
-			}{},
-			wantNil: []string{"Client"},
+			build: func(t *testing.T, mockCtrl *gomock.Controller) (tt testCase) {
+				tt.target = &struct {
+					Client someClient `notservice:"foo"`
+				}{}
+				tt.wantNil = []string{"Client"}
+				return
+			},
 		},
 		{
 			name: "unknown type non-nil",
-			target: &struct {
-				Client unknownClient `service:"foo"`
-			}{Client: unknownClient(struct{}{})},
-			wantNonNil: []string{"Client"},
+			build: func(t *testing.T, mockCtrl *gomock.Controller) (tt testCase) {
+				tt.target = &struct {
+					Client someClient `service:"foo"`
+				}{Client: someClient(struct{}{})}
+				tt.wantNonNil = []string{"Client"}
+				return
+			},
 		},
 		{
-			name:          "known type",
-			knownServices: []string{"foo"},
-			target: &struct {
-				Client knownClient `service:"foo"`
-			}{},
-			wantNonNil: []string{"Client"},
+			name: "known type",
+			build: func(t *testing.T, mockCtrl *gomock.Controller) (tt testCase) {
+				tt.knownServices = []string{"foo"}
+				tt.clientConfigs = []interface{}{
+					clientBuilderConfig{ClientConfig: gomock.Any()}.Get(t),
+				}
+				tt.target = &struct {
+					Client someClient `service:"foo"`
+				}{}
+				tt.wantNonNil = []string{"Client"}
+				return
+			},
 		},
 		{
-			name:          "default encodings",
-			knownServices: []string{"jsontest", "rawtest"},
-			target: &struct {
-				JSON json.Client `service:"jsontest"`
-				Raw  raw.Client  `service:"rawtest"`
-			}{},
-			wantNonNil: []string{"JSON", "Raw"},
+			name: "known type with struct field",
+			build: func(t *testing.T, mockCtrl *gomock.Controller) (tt testCase) {
+				tt.knownServices = []string{"foo"}
+				tt.clientConfigs = []interface{}{
+					clientBuilderConfig{
+						ClientConfig: gomock.Any(),
+						StructField: gomock.Eq(reflect.StructField{
+							Name:  "Client",
+							Type:  reflect.TypeOf((*someClient)(nil)).Elem(),
+							Index: []int{0},
+							Tag:   `service:"foo" thrift:"bar"`,
+						}),
+					}.Get(t),
+				}
+				tt.target = &struct {
+					Client someClient `service:"foo" thrift:"bar"`
+				}{}
+				tt.wantNonNil = []string{"Client"}
+				return
+			},
+		},
```

Comment:
can we add another test case with our "real" thrift values?  (Both multiplexed and enveloped)